cpp/cmake/3rdparty.cmake

```diff
@@ -0,0 +1,94 @@
+find_package(Git QUIET)
+
+if(NOT ${CODING_CONV_PREFIX}_3RDPARTY_DIR)
+  set(${CODING_CONV_PREFIX}_3RDPARTY_DIR 3rdparty)
+  set(${CODING_CONV_PREFIX}_3RDPARTY_DIR
+      3rdparty
+      PARENT_SCOPE)
+endif()
+
+# initialize submodule with given path
+function(bbp_init_git_submodule path)
+  if(NOT ${GIT_FOUND})
+    message(
+      FATAL_ERROR "git not found and ${path} submodule not cloned (use git clone --recursive)")
+  endif()
+  message(
+    STATUS "Fetching git submodule ${path}: running git submodule update --init --recursive ${path}"
+  )
+  execute_process(
+    COMMAND
+      ${GIT_EXECUTABLE} submodule update --init --recursive -- ${path}
```

Comment:
Can we make `--recursive` as an option instead of hardcoding it? e.g. if I am using Eigen as header only submodule, I want to initialise Eigen but not's its (large) submodules that are used for internal testing by Eigen. This is quite useful in neuron/coreneuron/nmodl use case.

Reply:
good point. I agree, maybe by default i would leave it on, but have an option to disable `--recursive` for a specific submodule

Reply:
The `git submodule update` command has many options that someone may want to use. So I would rather add a general `GIT_ARGS` option instead of something specific for the `--recursive` case:

```
bbp_git_submodule(foo GIT_ARGS --recursive)
```

Would that suit you @pramodk?

Reply:
thats also fine!

Reply:
`GIT_ARGS` option has been added (see [diff](https://github.com/BlueBrain/hpc-coding-conventions/compare/dce9a67...e145f99) of  633008372d42 + e145f99968e98e7c)